trieste/utils/objectives.py

```diff
@@ -126,6 +128,143 @@ def logarithmic_goldstein_price(x: TensorType) -> TensorType:
 """
 
 
+class MultiObjectiveTestProblem(ABC):
+    r"""Base class for test multi-objective test functions.
+    between a provided point and the closest point on the true pareto front.
+    """
+
+    @abstractmethod
+    def prepare_benchmark(self):
+        """
+        Evaluate Original problem
+        """
+
+    @abstractmethod
+    def gen_pareto_optimal_points(self, n: int):
+        """
+        Generate `n` pareto optimal points
+        """
+
+
+class VLMOP2(MultiObjectiveTestProblem):
+    """
+    The VLMOP2 function, typically evaluated over :math:`[-2, 2]^2`. See
+    :cite:`van1999multiobjective`  for details.
+    """
+
+    bounds = [[-2.0] * 2, [2.0] * 2]
+    dim = 2
+
+    def prepare_benchmark(self):
+        return vlmop2
+
+    def gen_pareto_optimal_points(self, n: int):
+        """
+        :cite: fonseca1995multiobjective
+        True pareto lies on -1/sqrt(2)-1/sqrt(2) and x1=x2
+        """
+        tf.debugging.assert_greater(n, 0)
+        _x = tf.linspace([-1 / tf.sqrt(2.0)], [1 / tf.sqrt(2.0)], n)
```

Comment:
underscores in local variable names isn't standard elsewhere in trieste, so for readability can we avoid them?

Reply:
okay